### PR TITLE
Replace custom tooltips with native one in vue components and views

### LINKED ISSUE #35654: [BITV]: Replace custom tooltips with native one for files_sharing
Replace custom tooltips with native one in view.

- [ ] Needs NC Vue release

## ``@nextcloud/vue``
- [x] https://github.com/nextcloud/nextcloud-vue/issues/3587
- [x] https://github.com/nextcloud/server/issues/35857
- [x] https://github.com/nextcloud/server/issues/35858
- [x] https://github.com/nextcloud/server/issues/35860
- [x] https://github.com/nextcloud/server/issues/35862

## Server

 - [ ]  SharingEntry.vue
 - [ ] SharingEntryInherited.vue
 - [ ] SharingEntryInternal.vue
 - [ ] SharingEntryLink.vue
 - [ ] SharingEntrySimple.vue
 - [ ] SharingInherited.vue
 - [ ] SharingTab.vue
 - [ ] SharesMixin.js
 - [ ] files_drop.js
 - [ ] public.js
 - [ ] sharedfilelist.js
 - [ ] share.js

This is a part of https://github.com/nextcloud/nextcloud-vue/issues/2503

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
Fixes https://github.com/nextcloud/server/issues/35654

- [x] Blocked: https://github.com/nextcloud/nextcloud-vue/pull/3569 (v7.3.0)

**Before example:**

Screenshot from 2022-12-08 13-46-45

Screenshot from 2022-12-09 11-37-19

**After example:**

Screenshot from 2022-12-09 09-24-32
Screenshot from 2022-12-09 09-24-55

Screenshot from 2022-12-09 11-38-06

## Summary

- [ ] Replace custom tooltips with native one in vue components and views
- [ ] Fix bug with wrong appearing notifications by setting password 
- [ ] Add aria-label as same as tooltip message

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_sharing/src/components/SharingEntryInternal.vue [hunk 1]
```diff
@@ -66,18 +66,18 @@ export default {
...
-		clipboardTooltip() {
+		actionButtonTooltip() {
 			if (this.copied) {
 				if (this.copySuccess) {
 					return ''
 				}
 				return t('files_sharing', 'Cannot copy, please copy the link manually')
 			}
-			return t('files_sharing', 'Copy to clipboard')
+			return t('files_sharing', 'Copy internal link to clipboard')
```
Comment (ChristophWurst) [inline apps/files_sharing/src/components/SharingEntryInternal.vue:80]: is this change necessary?

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 2]
```diff
@@ -566,18 +545,18 @@ export default {
...
-		clipboardTooltip() {
+		copyButtonTooltipMessage() {
 			if (this.copied) {
 				if (this.copySuccess) {
 					return ''
 				}
 				return t('files_sharing', 'Cannot copy, please copy the link manually')
 			}
-			return t('files_sharing', 'Copy to clipboard')
+			return t('files_sharing', 'Copy public link to clipboard')
```
Comment (ChristophWurst) [inline apps/files_sharing/src/components/SharingEntryLink.vue:559]: same here

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 3]
```diff
@@ -647,7 +626,9 @@ export default {
 						return true
 					} else {
 						this.open = true
-						OC.Notification.showTemporary(t('files_sharing', 'Error, please enter proper password and/or expiration date'))
+						const error = t('files_sharing', 'Error, please enter proper password and/or expiration date')
+						OC.Notification.showTemporary(error)
```
Comment (ChristophWurst) [inline apps/files_sharing/src/components/SharingEntryLink.vue:630]: doesn't this already trigger `showMessage` and therefore cause a *second* toast with the call one line below?

File: apps/files_sharing/src/utils/GeneratePassword.js [hunk 4]
```diff
@@ -39,10 +40,12 @@ export default async function() {
 		try {
 			const request = await axios.get(config.passwordPolicy.api.generate)
 			if (request.data.ocs.data.password) {
+				showSuccess(t('files_sharing', 'Password was successfully created'))
 				return request.data.ocs.data.password
 			}
 		} catch (error) {
 			console.info('Error generating password from password_policy', error)
+			showError(t('files_sharing', 'Error generating password from password_policy: {error}', { error }))
```
Comment (ChristophWurst) [inline apps/files_sharing/src/utils/GeneratePassword.js:48]:
```suggestion
			showError(t('files_sharing', 'Error generating password from password_policy'))
```

you would very likely show a technical error to an end user

File: apps/files_sharing/src/components/SharingEntryLink.vue (hunk 2 above)
Comment (JuliaKirschenheuter) [reply apps/files_sharing/src/components/SharingEntryLink.vue:559]: ok, i will take it back

File: apps/files_sharing/src/components/SharingEntryInternal.vue (hunk 1 above)
Comment (JuliaKirschenheuter) [reply apps/files_sharing/src/components/SharingEntryInternal.vue:80]: not necessary of coarse. i thought that specific info would be more useful

Comment (PVince81) [reply apps/files_sharing/src/components/SharingEntryInternal.vue:80]: since this is used by the aria-label for an action, it is necessary to have distinct texts for every link or action

otherwise people using screen readers who ask to "give me the list of actions on these page" will see multiple times the same text and won't know which to pick

File: apps/files_sharing/src/components/SharingEntryLink.vue (hunk 2 above)
Comment (PVince81) [reply apps/files_sharing/src/components/SharingEntryLink.vue:559]: see my comment about aria-label

File: apps/files_sharing/src/components/SharingEntryInternal.vue (hunk 1 above)
Comment (ChristophWurst) [reply apps/files_sharing/src/components/SharingEntryInternal.vue:80]: makes sense. then this is necessary

**Review by ChristophWurst — APPROVED** [review]

File: apps/files_sharing/src/components/SharingEntryInternal.vue [hunk 5]
```diff
@@ -10,11 +10,11 @@
 			</template>
 
 			<NcActionLink :href="internalLink"
-				:aria-label="t('files_sharing', 'Copy internal link to clipboard')"
+				:aria-label="copyLinkTooltip"
+				:title="copyLinkTooltip"
 				target="_blank"
 				:icon="copied && copySuccess ? 'icon-checkmark-color' : 'icon-clippy'"
 				@click.prevent="copyLink">
```
Comment (Pytal) [inline apps/files_sharing/src/components/SharingEntryInternal.vue:17]:
```suggestion
				@click.prevent="copyLink" />
```
**suggestion (non-blocking)**

File: apps/files_sharing/src/components/SharingEntryInternal.vue [hunk 6]
```diff
@@ -10,11 +10,11 @@
...
-				{{ clipboardTooltip }}
 			</NcActionLink>
```
Comment (Pytal) [inline apps/files_sharing/src/components/SharingEntryInternal.vue:18]:
```suggestion
```
**suggestion (non-blocking)**

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 7]
```diff
@@ -40,10 +40,10 @@
 			class="sharing-entry__copy">
 			<NcActionLink :href="shareLink"
 				target="_blank"
-				:aria-label="t('files_sharing', 'Copy public link to clipboard')"
+				:title="copyLinkTooltip"
+				:aria-label="copyLinkTooltip"
 				:icon="copied && copySuccess ? 'icon-checkmark-color' : 'icon-clippy'"
 				@click.stop.prevent="copyLink">
```
Comment (Pytal) [inline apps/files_sharing/src/components/SharingEntryLink.vue:46]:
```suggestion
				@click.stop.prevent="copyLink" />
```
**suggestion (non-blocking)**

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 8]
```diff
@@ -40,10 +40,10 @@
...
-				{{ clipboardTooltip }}
 			</NcActionLink>
```
Comment (Pytal) [inline apps/files_sharing/src/components/SharingEntryLink.vue:47]:
```suggestion
```
**suggestion (non-blocking)**

**Review by Pytal — APPROVED** [review]

**Comment by AndyScherzinger** [comment]
vue component has been released with fix, v.7.3.0

**Comment by PVince81** [comment]
- master 7.3.0 update: https://github.com/nextcloud/server/pull/35996
- stable25: TODO

**Comment by PVince81** [comment]
rebased and recompiled

**Comment by solracsf** [comment]
FYI this caused a regression affecting master and all versions (v26, v25): https://github.com/nextcloud/server/issues/38157
